app/controllers/users.js

```diff
@@ -82,3 +85,133 @@ exports.signin = (request, response, next) => {
+          response.status(200);
+          response.send({
+            paging: {
+              total: count,
+              limit,
+              offset
+            },
+            users: usersFiltered
+          });
+        });
+      } else {
+        return next(errors.defaultError('Error to trying to find all users'));
+      }
+    });
+  } else {
+    response.status(401);
+    response.send();
+  }
+};
+
+exports.signupAdministrator = (request, response, next) => {
+  const userLogged = request.user;
+  if (userLogged && userLogged.is_administrator) {
+    const saltRounds = 10;
+    const regexValidEmail = /^[A-Za-z0-9._%+-]+@wolox.com$/;
+    const regexAlphanumericPass = /^[a-z0-9]+$/i;
+
+    const newUser = request.body
+      ? {
+          firstName: request.body.firstName,
+          lastName: request.body.lastName,
+          email: request.body.email,
+          password: request.body.password,
+          is_administrator: true
+        }
+      : {};
+
+    if (newUser.password.length < 8 && !regexAlphanumericPass.test(newUser.password)) {
+      return next(errors.defaultError('La password debe ser alfanumerica y de 8 caracteres'));
+    }
+
+    if (!regexValidEmail.test(newUser.email)) {
+      return next(errors.defaultError('Debe ser un correo valido de wolox.com'));
+    }
+
+    bcrypt.hash(newUser.password, saltRounds).then(hash => {
+      newUser.password = hash;
+      userService
+        .getByEmail(newUser.email)
+        .then(user => {
+          if (user != null) {
+            user.isAdministrator = true;
+            user.save();
+            response.status(200);
+            response.send();
+          } else {
+            userService
+              .createUser(newUser)
+              .then(u => {
+                response.status(200);
+                response.send();
+              })
+              .catch(err => {
+                next(err);
+              });
+          }
+        })
+        .catch(err => {
+          next(err);
+        });
+    });
+  } else {
+    response.status(401);
+    response.send();
+  }
+};
+
+exports.invalidateAllSessions = (request, response, next) => {
+  const userLogged = request.user;
+  if (userLogged) {
+    userLogged.auth_code_validation = null;
+    userLogged.save();
+    response.status(200);
+    response.send();
+  } else {
+    response.status(401);
+    response.send();
+  }
+};
+
+exports.renewSession = (request, response, next) => {
+  const userLogged = request.user;
+  const accessToken = request.accessToken;
+  if (userLogged) {
+    sessionManager.generateAccessToken(userLogged.id).then(newAccessToken => {
+      userLogged.auth_code_validation = newAccessToken.authCode;
+      userLogged.save().then(success => {
+        response.status(200);
+        response.send({
+          accessToken: newAccessToken.accessToken,
+          refreshToken: newAccessToken.refreshToken
+        });
+      });
+    });
+  } else {
+    response.status(401);
```

Comment:
Another sendStatus (i'll mark you everyone so it's easier for you to find the ones that you need to replace)